src/song/AndSongFilter.cxx

```diff
@@ -54,9 +56,5 @@ AndSongFilter::ToExpression() const noexcept
 bool
 AndSongFilter::Match(const LightSong &song) const noexcept
 {
-	for (const auto &i : items)
-		if (!i->Match(song))
-			return false;
-
-	return true;
+	return std::any_of(items.begin(), items.end(), [&song](const auto &i) { return i->Match(song); });
```

Comment:
Yet another inverted check!

Reply:
Previously, the method returns true when there are exactly zero mismatches. (All items match)
With your code, the method returns true when there is at least one match. (Allows `n-1` mismatches)
See the difference?

A correct implementation could be:
```
return std::all_of(items.begin(), items.end(), [&song](const auto &i) { return i->Match(song); });
```
(i.e. `all_of` instead of `any_of`).